test/hotspot/jtreg/serviceability/jvmti/HeapMonitor/MyPackage/HeapMonitorStatObjectCorrectnessTest.java

```diff
@@ -46,6 +46,8 @@ private static void allocate() {
     emptyStorage();
 
     HeapMonitor.enableSamplingEvents();
+    // retire TLAB with GC, ensure sample happens, not assume TLAB size
```

Comment:
// Instead of relying on the allocation loop to fill and retire the TLAB, which might not happen,
// use System.gc() to retire the TLAB and ensure sampling happens